dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,587 @@
+
+                flag = counter > 0.0
+
+                parameter, parameter_cov, resid_rms = self.fit_point_source_transit(ha, vis.real, np.sqrt(var),
+                                                                                               flag=flag, fwhm=fwhm)
+            # Write to file
+            output_file = os.path.join(self.write_dir, "%s_csd_%d_%s.h5" % 
+                                      (src.lower(), csd, self.output_suffix))
+            self.logger.info("Writing output files...")
+            
+            with h5py.File(output_file, 'w') as handler:
+        
+                index_map = handler.create_group('index_map')
+                index_map.create_dataset('freq', data=data.index_map['freq'][:])
+                index_map.create_dataset('pol', data=np.string_(polstr))
+                index_map.create_dataset('time', data=data.time)
+                index_map.create_dataset('ra', data=ra)
+                index_map.create_dataset('ha', data=ha)
+        
+                dset = handler.create_dataset('vis', data=vis)
+                dset.attrs['axis'] = np.array(['freq', 'pol', 'ha'],dtype='S')
+                
+                dset = handler.create_dataset('weight', data=tools.invert_no_zero(var))
+                dset.attrs['axis'] = np.array(['freq', 'pol', 'ha'], dtype='S')
+                
+                dset = handler.create_dataset('count', data=counter.astype(np.int))
+                dset.attrs['axis'] = np.array(['freq', 'pol', 'ha'], dtype='S')
+                
+                if self.perform_fit:
+
+                    index_map.create_dataset('param', data='model_fitting')
+
+                    dset = handler.create_dataset('residual_noise', data=resid_rms)
+                    dset.attrs['axis'] = np.array(['freq', 'pol'], dtype='S')
+
+                    dset = handler.create_dataset('parameter', data=parameter)
+                    dset.attrs['axis'] = np.array(['freq', 'pol', 'param'], dtype='S')
+                
+                    dset = handler.create_dataset('parameter_cov', data=parameter_cov)
+                    dset.attrs['axis'] = np.array(['freq', 'pol', 'param', 'param'], dtype='S')
+                    
+                handler.attrs['source'] = src
+                handler.attrs['csd'] = csd
+                handler.attrs['instrument_name'] = self.correlator
+                handler.attrs['collection_server'] = subprocess.check_output(
+                    ["hostname"]).strip()
+                handler.attrs['system_user'] = subprocess.check_output(
+                    ["id", "-u", "-n"]).strip()
+                handler.attrs['git_version_tag'] = dias_version_tag
+            self.logger.info('File successfully written out.')
+
+    ###################################################
+    # auxiliary routines
+    ###################################################
```

Comment:
It seems like most of the auxiliary routines do not actually engage with the `self`. I am going to move them out of the class.